packages/test/test-utils/src/timeoutUtils.ts

```diff
@@ -20,12 +20,15 @@ class TestTimeout {
 
 	private static instance: TestTimeout = new TestTimeout();
 	public static reset(runnable: Mocha.Runnable) {
-		TestTimeout.clear();
+		TestTimeout.instance.clearTimer();
+		TestTimeout.instance = new TestTimeout();
 		TestTimeout.instance.resetTimer(runnable);
 	}
 
 	public static clear() {
+		// TODO: Check general setup here to see if we want tweaks.
```

Comment:
still want to think some more (or at least validate the changes don't break anything) about this file hence the TODO, but the current version definitely fixes the leaking aspect.

Reply:
it did break things and we had unit tests, which are now working :). resolved.